nixos/modules/services/misc/lms.nix

```diff
@@ -0,0 +1,234 @@
+        create and modify this directory.
+      '';
+    };
+
+    virtualHost = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      description = ''
+        Name of the nginx virtualhost to setup as reverse proxy. If null, do
+        not setup any virtualhost.
+      '';
+    };
+
+    listenAddress = mkOption {
+      type = types.str;
+      default = "127.0.0.1";
+      description = ''
+        The address on which to bind LMS.
+      '';
+    };
+
+    port = mkOption {
+      type = types.int;
+      default = 5082;
+      description = ''
+        The port on which LMS will listen to.
+      '';
+    };
+
+    theadCount = mkOption {
+      type = types.int;
+      default = 0;
+      description = ''
+        Number of threads to be used to dispatch http requests. 0 means auto
+        detect.
+      '';
+    };
+
+    listenbrainzApiBaseUrl = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = "https://api.listenbrainz.org";
+      description = ''
+        Listen brainz's root API url. If null, use default url.
+      '';
+    };
+
+    listenbrainzMaxSyncListenCount = mkOption {
+      type = types.int;
+      default = 1000;
+      description = ''
+        How many listens to retrieve when syncing. 0 disables sync.
+      '';
+    };
+
+    listenbrainzSyncPeriod = mkOption {
+      type = types.int;
+      default = 1;
+      description = ''
+        How often to resync listens. 0 disables sync.
+      '';
+    };
+
+    acousticbrainzApiBaseUrl = mkOption {
+      type = types.nullOr types.str;
+      default = null;
+      example = "https://acousticbrainz.org";
+      description = ''
+        Acoustic brainz's root API url. If null, use default url.
+      '';
+    };
+
+    loginThrottlerMaxEntries = mkOption {
+      type = types.int;
+      default = 10000;
+      description = ''
+        Max entries in the login throttler (1 entry per IP address. For IPv6,
+        the whole /64 block is used).
+      '';
+    };
+
+    enableSubsonicApi = mkOption {
+      type = types.bool;
+      default = true;
+      description = ''
+        Enable the Subsonic API.
+      '';
+    };
+
+    coverArtMaxSize = mkOption {
+      type = types.int;
+      default = 10;
+      description = ''
+        Max external cover file size in MBytes.
+      '';
+    };
+
+    coverArtCacheSize = mkOption {
+      type = types.int;
+      default = 30;
+      description = ''
+        Max cover cache size in MBytes.
+      '';
+    };
+
+    coverArtJpegQuality = mkOption {
+      type = types.int;
+      default = 75;
+      description = ''
+        JPEG quality for covers (range is 1-100).
+      '';
+    };
+
+    user = mkOption {
+      type = types.str;
+      default = "lms";
+      description = ''
+        User account under which LMS runs. You have to create the user
+        yourself if you specify a different user from the default.
+      '';
```

Comment:
```suggestion
      description = ''
        User account under which LMS runs. If left as the default value this user will automatically be created on system activation, otherwise the sysadmin is responsible for ensuring the user exists before the lms service starts.
      '';
```